### PR TITLE
Update DG - Implementations

### LINKED ISSUE #516: Update DG - Implementations
* Update diagrams and descriptions for implementations of `login`, `logout`, `customer add` based on latest code quality changes

### PR DESCRIPTION
**Pull request by zhonghan721**
Closes #516 

Update implementations for `login`, `logout`, `customer add`

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -562,16 +564,17 @@ The sequence of the `logout` command is as follows:
 
 1. The user inputs the `logout` command.
 2. A new `userLogoutCommand` is created and checks whether the user is currently logged out
-   by calling `Model#getUserLoginStatus()`.
-3. If the user is currently logged in, the `userLogoutCommand` calls `Model#setLogoutSuccess()`,
+   by calling `Model#getUserLoginStatus`.
+3. If the user is currently logged in, the `userLogoutCommand` calls `Model#setLogoutSuccess`,
    changing the login status to false and restricting the user access to most commands.
-4. The `userLoginCommand` also calls `Model#updateFilteredPersonList()` to hide the list of customers.
+4. The `userLoginCommand` also calls `Model#clearFilteredDeliveryList` and `Model#clearFilteredCustomerList`
```
Comment (B-enguin) [inline docs/DeveloperGuide.md]: Is this supposed to be userLogoutCommand?

**Review by B-enguin — COMMENTED** [review]
Just some general comments

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zhonghan721) [reply docs/DeveloperGuide.md]: Good catch, thanks!

**Review by B-enguin — APPROVED** [review]
LGTM

**Review by jianyangg — APPROVED** [review]
LGTM

**Pull request merged**